Makefile

```diff
@@ -810,13 +810,28 @@ BINARYNAME=$(TARNAME)-$(PLATFORM)-$(ARCH)
 endif
 BINARYTAR=$(BINARYNAME).tar
 # OSX doesn't have xz installed by default, http://macpkg.sourceforge.net/
-XZ=$(shell which xz > /dev/null 2>&1; echo $$?)
+HAS_XZ ?= $(shell which xz > /dev/null 2>&1; [[ $$? = 0 ]] && echo 1 || echo 0)
+# Supply SKIP_XZ=1 to explicitly skip .tar.xz creation
+SKIP_XZ ?= 0
```

Comment:
what do I do with that then? we need to be able to differentiate between "have xz", "have xz but asked to skip", "don't have xz" and "don't have xz but have to skip". There needs to be a hard fail as well as a soft skip.

Reply:
Just to keep the churn low. It we compute the value of `XZ` we can keep the later guards as is.
Like https://github.com/nodejs/node/blob/ba2647a559439b6c8a8ec6573cba8aee2ebed59a/Makefile#L961
etc.

Reply:
I'm not concerned about churn, I'd rather aim for grokkability in Makefile since it's already full of magic. So is your suggestion that, with the `check-xz` as it is in this PR (i.e. with HAS_XZ and SKIP_XZ intact), that we roll up the blocks surrounding the `xz` call under the `XZ` boolean?

`ifeq ($(HAS_XZ) $(SKIP_XZ), 1 0)` is a little esoteric, so perhaps that'd be a good change for grokkability sake.

Reply:
Yes. That "if" is very specific. If we restore `XZ` but now it's cumputed as `($(HAS_XZ) $(SKIP_XZ), 1 0)` (even comment that it means "we have a working `xz` available") the lower code blocks will be simpler to grok.